### PR TITLE
Reduced GroundingDataset annotation unpacking speed

### PR DESCRIPTION
**Pull request by Lornatang**
Reduce IO latency caused by multiple unpacking when faced with a large number of annotations

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary  
Improved dataset handling by reorganizing captions for cleaner and more efficient processing. 🛠️

### 📊 Key Changes  
- Refactored code to directly assign `caption = img["caption"]` before using it in label extraction logic.  
- Simplified and clarified how category names (`cat_name`) are pulled from captions during annotation processing.

### 🎯 Purpose & Impact  
- **Purpose**: To enhance code readability and maintainability by removing redundancy in caption handling.  
- **Impact**: This change ensures smoother, more efficient dataset operations, which is great for both developers working on the code and users expecting swift and reliable annotation processing. 🚀

### CODE REVIEW

File: ultralytics/data/dataset.py [hunk 1]
```diff
@@ -323,7 +323,9 @@ def get_labels(self):
                 if box[2] <= 0 or box[3] <= 0:
                     continue
 
-                cat_name = " ".join([img["caption"][t[0] : t[1]] for t in ann["tokens_positive"]])
+                caption = img["caption"]
+                tokens_positive = ann["tokens_positive"]
+                cat_name = " ".join([caption[t[0]: t[1]] for t in tokens_positive])
```
Comment (Laughing-q) [inline ultralytics/data/dataset.py:328]: @Lornatang How about let's keep the original `ann["tokens_positive"]` since the for loop would access it once no?

Comment (Lornatang) [reply ultralytics/data/dataset.py:328]: Some json files may contain this field multiple times.
For example, YOLO_World official cc3m dataset json [cc3m_lite](https://huggingface.co/wondervictor/YOLO-World/blob/main/cc3m_pseudo_annotations.json).

Comment (Laughing-q) [reply ultralytics/data/dataset.py:328]: but here we are using `ann["tokens_positive"]` once, not like we are looping `img["caption"]` multiple times.
```python
cat_name = " ".join([img["caption"][t[0] : t[1]] for t in ann["tokens_positive"]])
```

Comment (Lornatang) [reply ultralytics/data/dataset.py:328]: Ok, I see. This change may work if more GroundingDataset types are supported in the future, thanks.

Comment (Laughing-q) [reply ultralytics/data/dataset.py]: @Lornatang ok reverted the `ann["tokens_positive"]` in https://github.com/ultralytics/ultralytics/pull/18382/commits/052cb33a944f6caf04b9dc0abe3536fc1acaf49c

**Review by Laughing-q — APPROVED** [review]

**Review by glenn-jocher — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
🎉 PR Merged! Huge thanks to @Lornatang for leading this improvement, with valuable insights from @Laughing-q and @glenn-jocher. Your collaborative effort to simplify and refine dataset handling ensures faster, cleaner, and smarter processes for everyone. 🚀✨  

As Leonardo da Vinci once said, *"Simplicity is the ultimate sophistication."* This work embodies that spirit by making the code more elegant and impactful. Your contributions today will ripple out into countless projects, empowering developers and creators alike. Bravo! 🙌

**Pull request merged**